initial release

generationmake opened the pull request.

generationmake committed:
added code

generationmake committed:
added code to copy from CanMsg to CanardFrame, somehow compiling

generationmake commented:
Hi @aentinger
Could you take a look at this?
It seems like i am having a problem with the linker and heap.

aentinger commented:
a) please add me as a collaborator to this repository.
b) what linker error do you see ... this one:
```bash
/tmp/arduino/sketches/C84D9B71B666A469DC3CD95253872BF4/core/core.a(USB.cpp.o): In function `__USBStart()':
/home/alex/Arduino/hardware/arduino-git/renesas/cores/arduino/usb/USB.cpp:328: undefined reference to `tusb_rusb2_set_irqnum'
collect2: error: ld returned 1 exit status
```
?

aentinger committed:
Fix: Cyphal/CAN uses 29-Bit (also known as extended) CAN IDs.

generationmake commented:
```
/home/bernhard/.arduino15/packages/arduino/tools/arm-none-eabi-gcc/7-2017q4/bin/../lib/gcc/arm-none-eabi/7.2.1/../../../../arm-none-eabi/bin/ld: section .stack_dummy VMA [0000000020007b00,0000000020007eff] overlaps section .heap VMA [0000000020005cb0,0000000020007caf]
collect2: error: ld returned 1 exit status

```

aentinger committed:
Fix: half required heap size for Cyphal.
This is necessary because the RA4M1 MCU on Arduino Uno R4 WiFi/Minima only comes with 32 kB of RAM. Compare this to a 256 kB of a Raspberry Pi RP2040.

aentinger commented:
"Fixed" the linker error, for details please see commit comment: https://github.com/generationmake/cyphal_on_R4/pull/1/commits/5d446f5eec68fe335e967f1609339176def972aa .

aentinger committed:
Patching it up to work.

generationmake committed:
corrected WDT

generationmake committed:
replaced estop with input
